### PR TITLE
rename functions involving side effect

### PR DESCRIPTION
**Pull request by iblis17**
```
libsvm_set_verbose     -> libsvm_set_verbose!
libsvm_set_num_threads -> libsvm_set_num_threads!
set_num_threads        -> set_num_threads!
```

### CODE REVIEW

**Comment by barucden** [comment]
I am not sure about the naming. Sure, the methods do have side effects, but they don't modify their arguments. I.e. I would expect that function `libsvm_set_verbose!(v)` would modify `v`. Of course, that might be just me...

File: src/LIBSVM.jl [hunk 1]
```diff
@@ -347,10 +348,7 @@ function svmtrain(
     (nodes, nodeptrs) = instances2nodes(X)
     problem = SVMProblem(Int32(size(X, 2)), pointer(idx), pointer(nodeptrs))
 
-    # Validate the given parameters
-    libsvm_check_parameter(problem, param)
```
[the remark this answers is not in the document]
Comment (iblis17) [reply src/LIBSVM.jl:351]: No, I think that's an accident :facepalm:

[the remark this answers is not in the document]
Comment (iblis17) [reply src/LIBSVM.jl:351]: ah, it's placed in the GC block in the following line.

**Comment by iblis17** [comment]
> Sure, the methods do have side effects, but they don't modify their arguments.

Yeah, the c function global states is a tricky one.
I'm fine with original naming.
I just re-read this: https://docs.julialang.org/en/v1/manual/style-guide/#bang-convention

**Pull request closed without merging**